src/main/java/com/example/user_management/api/controller/UserController.java

```diff
@@ -30,7 +29,8 @@ public ResponseEntity<GetUserResponseModel> getUserById(@PathVariable Long userI
     }
 
     @GetMapping("/")
-    public List<User> getAllUsers() {
-        return userService.getAllUsers();
+    public ResponseEntity<Page<User>> getAllUsers(@RequestParam int page, @RequestParam int size) {
```

Comment:
Should this api contract be like ResponseEntity<Page<GetUserResponseModel>> ? Why we return User entity with all fields from API ? for instance password should never be exposed via API.